### PR TITLE
Bazel Build Tool

### PR DESCRIPTION
**Pull request by WhiteBoardDev**
- building a PAR file so the application can be ran as a single binary
- avoids pip install commands
- testing working with bazel
- documentation upgrade

### CODE REVIEW

File: .gitignore [hunk 1]
```diff
@@ -1,4 +1,8 @@
-build
 .idea
 *.pyc
 venv
+bazel-bin
```
Comment (cnishina) [inline .gitignore]: nit: bazel-* would probably cover it all.

File: BUILD [hunk 2]
```diff
@@ -0,0 +1,32 @@
...
+load("@subpar//:subpar.bzl", "par_binary")
+load("@my_deps//:requirements.bzl", "requirement")
+
+par_binary (
+    name="rss_downloader",
+    srcs=[
+        'rss_downloader.py'
+    ],
+    deps = [
+    requirement("appdirs"),
```
Comment (cnishina) [inline BUILD]: add a tab to `requirement("` as well as other requirements below

File: BUILD [hunk 3]
```diff
@@ -0,0 +1,32 @@
...
+    requirement("feedparser"),
+    requirement("packaging"),
+    requirement("pyparsing"),
+    requirement("requests"),
+    requirement("schedule"),
+    requirement("six")
```
Comment (cnishina) [inline BUILD]: requirement("six"),

comma at the end is okay

File: BUILD [hunk 4]
```diff
@@ -0,0 +1,32 @@
+package(default_visibility = ["//visibility:public"])
+load("@subpar//:subpar.bzl", "par_binary")
+load("@my_deps//:requirements.bzl", "requirement")
+
+par_binary (
+    name="rss_downloader",
+    srcs=[
+        'rss_downloader.py'
```
Comment (cnishina) [inline BUILD]: either use single quotes or double quotes. stay consistent.

also, comma at end of everything: `"rss_downloader.py",`

File: BUILD [hunk 5]
```diff
@@ -0,0 +1,32 @@
+package(default_visibility = ["//visibility:public"])
+load("@subpar//:subpar.bzl", "par_binary")
+load("@my_deps//:requirements.bzl", "requirement")
+
+par_binary (
+    name="rss_downloader",
```
Comment (cnishina) [inline BUILD]: add spaces between equal:

```
    name = "rss_downloader",
    srcs = [
```

File: BUILD [hunk 6]
```diff
@@ -0,0 +1,32 @@
...
+    requirement("packaging"),
+    requirement("pyparsing"),
+    requirement("requests"),
+    requirement("schedule"),
+    requirement("six")
+    ]
+)
+
+py_test(
+    name='test_rss_downloader',
```
Comment (cnishina) [inline BUILD]: nit: use either single or double quote.

File: BUILD [hunk 7]
```diff
@@ -0,0 +1,32 @@
...
+    srcs=[
+        'test_cases.py',
+
+    ],
+    data = [ 'test_responses' ],
```
Comment (cnishina) [inline BUILD]: is there a data file?

File: BUILD [hunk 8]
```diff
@@ -0,0 +1,31 @@
...
+    ],
+    deps = [
+      requirement("appdirs"),
+      requirement("feedparser"),
+      requirement("packaging"),
+      requirement("pyparsing"),
+      requirement("requests"),
+      requirement("schedule"),
+      requirement("six")
+    ]
```
Comment (cnishina) [inline BUILD]: comma at end of everything. `],`

File: BUILD [hunk 9]
```diff
@@ -0,0 +1,31 @@
...
+)
+
+py_test(
+    name='test_rss_downloader',
+    srcs=[
+        'test_cases.py'
```
Comment (cnishina) [inline BUILD]: `'test_cases.py',`

File: BUILD [hunk 10]
```diff
@@ -0,0 +1,31 @@
...
+    ],
+    data = [ 'test_responses' ],
+    deps=[
+        ':rss_downloader'
```
Comment (cnishina) [inline BUILD]: comma at end of `':rss_downloader',`

File: BUILD [hunk 11]
```diff
@@ -0,0 +1,31 @@
...
+      requirement("six")
+    ]
+)
+
+py_test(
+    name='test_rss_downloader',
+    srcs=[
+        'test_cases.py'
+    ],
+    data = [ 'test_responses' ],
```
Comment (cnishina) [inline BUILD]: is there a data file called `'test_responses'`?

File: BUILD [hunk 12]
```diff
@@ -0,0 +1,31 @@
...
+    deps=[
+        ':rss_downloader'
+    ],
+    main = 'test_cases.py'
```
Comment (cnishina) [inline BUILD]: comma at end of line

File: BUILD [hunk 13]
```diff
@@ -0,0 +1,31 @@
...
+    ]
+)
+
+py_test(
+    name='test_rss_downloader',
+    srcs=[
+        'test_cases.py'
+    ],
+    data = [ 'test_responses' ],
+    deps=[
```
Comment (cnishina) [inline BUILD]: space between =

File: BUILD (hunk 11 above)
Comment (WhiteBoardDev) [reply BUILD]: yes. A data folder called test_responses contains mock RSS feed responses. During testing, an http server is spun up and then configured to use the test_responses.

**Pull request merged**